[Feat] #61 로그인/회원가입 UI 최종 디자인 구현

thisminn opened the pull request:
## 🔗 관련 이슈

- close #61 

<br>

## 📌 작업사항

- 카카오 로그인 전 약관동의 모달 페이지 구현
- 약관동의 markdown 파일 생성
- signupPages background 이미지 및 캐릭터 삽입
- userInfoForm -> input field 추가(3개->4개(phoneNumber)) 

<br>

## 🗒️ 참고사항

.md 파일들이 피그마에서 주신 내용과 같이 정갈하게 줄바꿈이 안되고 있습니다 🥲 이 부분 계속 수정해보겠습니다

<br>

## 📸 스크린샷

https://github.com/user-attachments/assets/dfd24d0a-8f65-476d-baca-f3ca600a9250



<!-- UI 변경 사항이 있다면, 관련 스크린샷을 첨부해주세요. -->

<br>

## ✅ 체크리스트

<!-- PR을 제출하기 전에 확인해야 할 항목들 -->

- [x] 로컬에서 빌드 및 테스트 완료
- [x] Label 지정 완료
- [x] 리뷰어 지정 완료
- [ ] 코드 리뷰 반영 완료
- [ ] 문서화 필요 여부 확인

thisminn committed:
feat: signup 페이지 배경 추가
intro page 캐릭터 svg 추가
userInfoForm - inputcontainer 개수 추가 및 value 추가(phoneNumber)

thisminn committed:
feat: 약관동의 모달 구현

thisminn committed:
feat: 약관 md 문서 생성

thisminn committed:
feat: markdown 문서 생성 및 연결

thisminn committed:
feat: 약관 확인 후 navigate 시 check 상태 유지시키는 기능 구현

thisminn committed:
feat: 약관동의 모달 버튼 추가

thisminn committed:
refactor: SignupPage 파일명 수정

thisminn committed:
refactor: 미사용 훅 선언문 제거

yezzan9 commented on src/components/ui/Button/RadioButton/RadioButtonContainer.tsx:
```diff
@@ -55,13 +55,13 @@ const RadioButtonContainer = ({
 
   return (
     <div className={containerClass}>
-      <label className='flex items-center text-[12px] font-bold leading-[14px] tracking-[-0.12px] mb-[6px]'>
```
여기 tracking 앞에 공백이 한칸 추가된 것 같아요!

yezzan9 commented on src/components/ui/Button/RadioButton/RadioButtonContainer.tsx:
```diff
@@ -55,13 +55,13 @@ const RadioButtonContainer = ({
         )}
       </label>
-      <div className='flex justify-between'>
+      <div className='flex  items-center h-[20px] gap-x-20'>
```
여기도 items-center 앞에 공백이 추가되었어요!

yezzan9 commented on src/components/ui/Button/KakaoLoginButton.tsx:
```diff
@@ -1,28 +1,138 @@
+
+      <ModalSheet isOpen={isModalOpen} onClose={handleCloseModal}>
+        <ul className='space-y-2 mb-6'>
+          {[
+            '[필수] 이용약관',
+            '[필수] 개인정보의 수집 및 이용에 대한 동의',
+            '[선택] 개인정보의 수집 및 이용에 대한 동의',
+          ].map((label, idx) => (
```
배열을 매핑하는 곳에 직접 작성하기보다는 미리 선언을 하고, 해당 배열을 가져오는 편이 가독성이 좋을 것 같아요 !

yezzan9 commented on src/components/ui/Button/KakaoLoginButton.tsx:
```diff
@@ -1,28 +1,138 @@
+                  onClick={(e) => {
+                    e.stopPropagation();
+
+                    navigate(
+                      `/terms/${['required-terms', 'required-privacy', 'optional-privacy'][idx]}?${searchParams.toString()}`,
+                    );
```
여기 `['required-terms', 'required-privacy', 'optional-privacy']` 부분은 
```js
[
    '[필수] 이용약관',
    '[필수] 개인정보의 수집 및 이용에 대한 동의',
    '[선택] 개인정보의 수집 및 이용에 대한 동의',
]
```
와 항상 짝을 이루며 사용되는 값이니, 두 값을 함께 객체로 관리하면 더 편할 것 같아요 !

yezzan9 replied:
근데 이 부분에서 searchParams는 어떤 용도로 사용되는 값인 건가요??

yezzan9 commented on src/components/ui/Modal/ModalSheet.tsx:
```diff
@@ -0,0 +1,54 @@
+  return (
+    <ModalPortal>
+      {/* Overlay */}
+      <div
+        className='fixed inset-0 z-50 flex items-end justify-center'
+        onClick={closeOnOverlayClick ? onClose : undefined}
+      >
```
`closeOnOverlayClick`은 모달 바깥 부분 클릭 시 onClose 함수 호출 여부를 판단하는 값이 맞을까용?-?
모달의 경우 모달 바깥 부분을 클릭했을 때 닫히는 게 일반적인 동작이니, 기획적으로 배경을 클릭해도 닫히지 않아야하는 경우가 있는 게 아니라면 불필요한 값인 것 같습니다!
닫히지 않아야하는 경우에도 배경div에 onClick 값을 주지 않으면 간단하게 구현이 가능하구요!

yezzan9 commented on src/components/ui/UserInfoForm.tsx:
```diff
@@ -37,6 +41,14 @@ const UserInfoForm = ({
     }
   };
 
+  const formatPhoneNumber = (value: string) => {
+    const digits = value.replace(/\D/g, ''); // 숫자만 남김
+    if (digits.startsWith('010')) {
+      return digits.replace(/^(\d{3})(\d{4})(\d{0,4}).*/, '$1-$2-$3').replace(/-$/, ''); // 마지막 - 제거
+    }
+    return digits;
+  };
```
정규식을 이용하여 포맷팅도 잘 구현해주셨네요!

궁금한 점이 있는데, 이렇게 구현하면 데이터를 저장할 때도 문자열로 저장하는 건가요?
물론 문자열로 저장할 때에도 정규식을 이용하여 간편하게 유효성 검사가 가능하지만, 개인적으로 숫자값의 경우는 number 타입으로 관리하는 것이 좋다고 생각합니다 !!
백엔드분들과 한번 얘기해보시고, 사용자에게 보여주는 것(하이픈 포함)과 실제 값(하이픈 제외 숫자만)을 나누어서 관리하도록 수정하는 것도 좋은 방법일 것 같아요 !!

yezzan9 commented on src/features/signup/SignupGuardianForm.tsx:
```diff
@@ -10,20 +11,24 @@ const SignupGuardianForm = ({ onNext }: Props) => {
+  const [phoneNumber, setPhoneNumber] = useState('');
 
-  const isFormIncomplete = !gender || !birthDate || !name;
+  const isFormIncomplete =
+    !gender || !birthDate || !name || phoneNumber.replace(/\D/g, '').length !== 11;
```
위의 코멘트 부분은 요런 부분에서도
```js
phoneNumber.length !== 11
```
이렇게 바로 확인할 수 있으니 연산이 줄어들어 더 효율적이지 않을까 싶어요 !
첫 세자리가 010인지 확인하는 유효성 검사도 추후에 추가되면 좋을 것 같습니다 ~~

yezzan9 commented on src/pages/auth/terms/TermsPage.tsx:
```diff
@@ -0,0 +1,61 @@
+import requiredPrivacy from '@/assets/terms/required-privacy.md?raw';
+import optionalPrivacy from '@/assets/terms/optional-privacy.md?raw';
+
+const TERMS_MAP: Record<string, string> = {
+  'required-terms': requiredTerms,
+  'required-privacy': requiredPrivacy,
+  'optional-privacy': optionalPrivacy,
+};
+const MESSAGES_MAP: Record<string, string> = {
+  'required-terms': '치카해 서비스 이용약관',
+  'required-privacy': '[필수] 개인정보 수집 및 이용에 대한 동의',
+  'optional-privacy': '[선택] 개인정보 수집 및 이용에 대한 동의',
+};
```
오잉 여기에 이미 선언을 해주셨군요 ,,
이걸 전역으로 선언하고, 아까 위쪽의 코멘트 부분에서 사용하면 좋을 것 같아요!

그리고 key값이 완전히 동일해서, 이렇게 분리해서 관리하기보다는 객체로 한번에 관리하는 편이 좋을 것 같아요!!

yezzan9 reviewed (commented):
코드에 대한 코멘트 외에도 몇가지 코멘트 남겨드렸으니 확인하시고 답변 주시면 감사하겠습니다👍🏻

 
- 영상을 확인해보면 선택약관에 체크를 해제했을 때에도 다음으로 넘어가는 버튼이 비활성화 되는데, 이러면 선택이 아니게 되지 않나요? 선택 약관의 체크 여부와는 관계 없이 필수 약관의 값만 확인하도록 수정이 필요해보입니다!
  <img width="257" height="168" alt="Image" src="https://github.com/user-attachments/assets/1ab4bd4b-e622-44fd-b430-88b645e4b566" />

- 이미 로그인을 한 상태에서 다시 `/login` 으로 접근이 가능한 것 같은데, 로그인을 한 상태라면 `/login`에 접근 시 홈으로 리다이렉트 되는 식으로 접근을 막는 것이 좋을 것 같아요 !

- 영상을 통으로 올려주시는 것보다는 약관 동의 / 14세 이상 가입 / 14세 미만 가입 이런 식으로 경우를 나눠서 올려주시면 원하는 부분을 빠르게 찾고 확인할 수 있어서 좋을 것 같아요 ~~

- > .md 파일들이 피그마에서 주신 내용과 같이 정갈하게 줄바꿈이 안되고 있습니다 🥲 이 부분 계속 수정해보겠습니다

  이 부분은 어떤 걸 말씀하시는 걸까요?

- 모달을 보면 드래그 동작을 기대하는 UI인 것 같은데, 기능적으로 구현이 안 되어있는 것 같아요! 드래그는 따로 고려하지 않기로 기획한 건가요?-?

nabbang6 commented on src/components/ui/UserInfoForm.tsx:
```diff
@@ -37,6 +41,14 @@ const UserInfoForm = ({
+      return digits.replace(/^(\d{3})(\d{4})(\d{0,4}).*/, '$1-$2-$3').replace(/-$/, ''); // 마지막 - 제거
+    }
+    return digits;
+  };
```
현재 전화번호 부분은 잘 포맷팅해주셨는데, 
스웨거 확인해보니 생년월일 부분은 `2017.08.25` 형식이 아니라 `2017-08-25` 이런 형식으로 데이터를 보내줘야 하는 것 같아요! 생년월일에도 변환 함수 적용해주시면 좋을 것 같습니당

<img width="381" height="164" alt="image" src="https://github.com/user-attachments/assets/52943b64-cbc4-41b2-86a9-11eac071df6b" />

thisminn replied:
반영했습니다!

nabbang6 reviewed (commented):
다 잘 구현해주신 것 같은데, 지금 스웨거에 변경사항이 조금 생긴 것 같아요! 
`gender` 값이 string이 아니라 boolean으로 받아오게끔 바뀐 것 같아서, 요부분도 한번 확인해주시면 좋을 것 같습니다 ㅜ_ㅜ ui 내용은 아니라서 #57 에서 수정해주셔도 될 것 같아요!

고생하셨습니다!! 👍👍

thisminn committed:
Merge branch 'develop' into feat/#61-signup-design

thisminn commented on src/components/ui/Nav/GlobalTopNav.tsx:
```diff
@@ -29,9 +29,9 @@ const GlobalTopNav = ({
         flex h-14 px-4
         items-center
         bg-bg-tertiary-gray
```
적용했습니다!

eun-seoo reviewed (approved):
민성님 수고 많으셨습니다~!
전체적으로 깔끔하게 구현해주셨고, 예진님과 나영님께서 적재적소에 리뷰를 잘 달아주셔서
제가 따로 더 드릴 피드백은 없을 것 같아요 :)
기존에 받으신 코멘트들만 반영해주시면 바로 머지 가능할 것 같습니다.
미리 어프루브 해놓겠습니다~! 🙌

thisminn committed:
feat: 로그인 여부에 따른 loginPage 네비게이션 조건 추가
topNav 배경색 변경
phoneNumber 유효성 검사 반영(010-)
약관동의 modal complete 기준 변경
생년월일 변환 함수 작성

thisminn commented on src/components/ui/UserInfoForm.tsx:
```diff
@@ -110,7 +110,7 @@ const UserInfoForm = ({
-            placeholder='전화번호를 입력해 주세요.'
+            placeholder='010-'
             variant='formBottom'
             value={phoneNumber}
```
아 이 부분은 기능 추가해서 함수로 작성해뒀습니다(010 자동으로 입력되어 시작하는)
그래서 사용자가 헷갈릴 수도 있을 것 같아서 뒷번호 8자리만 입력하라는 의도로 placeholder를 작성해뒀습니다 이 부분도 디자인 측이랑 논의 후 추가 수정 해서 pr에 추가하겠습니다!
